### PR TITLE
Support onbuild instruction

### PR DESCRIPTION
**Pull request by mesaglio**
Hi!

With this [onbuild instruction](https://docs.docker.com/engine/reference/builder/#onbuild) support we can add behavior to the apko images when we use them as base images.

### CODE REVIEW

**Comment by mesaglio** [comment]
Hi! @imjasonh @kaniini, can you help me with this pr?

File: pkg/build/types/build_option.go [hunk 1]
```diff
@@ -41,6 +41,7 @@ type BuildOption struct {
 	Environment map[string]string `yaml:"environment,omitempty"`
 
 	Entrypoint ImageEntrypoint `yaml:"entrypoint,omitempty"`
+	OnBuild    []string        `yaml:"on_build,omitempty"`
```
Comment (kaniini) [inline pkg/build/types/build_option.go:44]:
```suggestion
	OnBuild    []string        `yaml:"on-build,omitempty"`
```

File: pkg/build/types/image_configuration.go [hunk 2]
```diff
@@ -205,4 +205,11 @@ func (ic *ImageConfiguration) Summarize(logger log.Logger) {
 			logger.Printf("      %s: %s", k, v)
 		}
 	}
+
+	if len(ic.OnBuild) > 0 {
+		logger.Printf("  on_build:")
```
Comment (kaniini) [inline pkg/build/types/image_configuration.go:210]:
```suggestion
		logger.Printf("  on-build:")
```

File: pkg/build/types/image_configuration.go [hunk 3]
```diff
@@ -205,4 +205,11 @@ func (ic *ImageConfiguration) Summarize(logger log.Logger) {
...
+		for _, onBuildInstruccion := range ic.OnBuild {
```
Comment (kaniini) [inline pkg/build/types/image_configuration.go:211]:
```suggestion
		for _, onBuildInstruction := range ic.OnBuild {
```

File: pkg/build/types/image_configuration.go [hunk 4]
```diff
@@ -205,4 +205,11 @@ func (ic *ImageConfiguration) Summarize(logger log.Logger) {
...
+			logger.Printf("    - %s", onBuildInstruccion)
```
Comment (kaniini) [inline pkg/build/types/image_configuration.go:212]:
```suggestion
			logger.Printf("    - %s", onBuildInstruction)
```

**Review by kaniini — CHANGES_REQUESTED** [review]
In general, I don't think we should encourage doing this, but it seems fine.

**Comment by imjasonh** [comment]
I'm a little curious to learn more about your use case. Do you have an example scenario in mind? I thought `ONBUILD` instructions in Dockerfiles were deprecated and recommended against for quite some time, but maybe I'm mistaken.

**Comment by mesaglio** [comment]
Hello @imjasonh, I did not find in the official docker documentation that the onbuild instructions had been deprecated but I will look for it.

One use case we are looking at is having a distroless image for production and another in the development stage. Seeking that the environment is not so different, we think that the development environment is the production environment + busybox + compilation.
I thought some dockerfile with this style
```dockerfile
FROM prod_img:latest
# this can be added with melango + apko
RUN apk add busybox
ONBUILD COPY . /app
WORKDIR /app
ONBUILD CMD ["python3", "-m", "app"]
```

Final Dockerfile that devs can use
```dockerfile
FROM dev_img:latest
```

**Comment by mesaglio** [comment]
This have no sense 😢

**Comment by mfortin** [comment]
@mesaglio did you find an alternative ?
I was looking for this feature too

**Comment by mesaglio** [comment]
No I dont. But I understand that `ONBUILD` instruction is in [Docker image spec](https://github.com/moby/docker-image-spec/blob/v1.2.0/v1.2.md) no in [OCI image spec](https://specs.opencontainers.org/image-spec/config/?v=v1.0.1)

**Pull request closed without merging**